trail_detection_node/trail_detection_node/visualizer_v2_trailDetectionNode.py

```diff
@@ -0,0 +1,413 @@
+import torch
+from .model_loader import FCN8s, FCN32s, PSPNet
+from PIL import Image as ImagePIL
+from torchvision import transforms
+import cv2
+import os
+
+import rclpy
+from rclpy.node import Node
+from geometry_msgs.msg import PoseStamped
+from sensor_msgs.msg import Image, PointCloud2
+import sensor_msgs_py.point_cloud2 as pc2
+import numpy as np
+import math
+from cv_bridge import CvBridge
+
+import message_filters
+from scipy.ndimage import grey_erosion
+'''
+    The transformation matrix as well as the coordinate conversion and depth estimation functions are copied from human_detection_node
+'''
+camera_transformation_k = np.array([
+    [628.5359544,0,676.9575694],
+    [0,627.7249542,532.7206716],
+    [0,0,1]])
+
+rotation_matrix = np.array([
+    [-0.007495781893,-0.0006277316155,0.9999717092],
```

Comment:
already defined in the node file.  Here too? Why?  Also defined elsewhere in trailbot repo.

Reply:
please change the structure so there is no duplicate code for the same node.  Make visualization an option that can be turned on and off.  This is fine to do in code as opposed to with a config parameter, although a config parameter would be better.